### ISSUE TITLE
HTTP Interface - "Internal Server Error"

### ISSUE THREAD

**Issue by Slevinox** [mistio/mist-ce#938]
Hello,

I installed following the README but after starting all containers.

All I got connecting to "https://192.168.100.XX" is : 
"Internal server Error"

Here are my specs : 
- OS : CentOS Linux release 7.5.1804 (Core)
- CPU : 8 Cores, Intel(R) Xeon(R) CPU E5540  @ 2.53GHz
- RAM : 
total        used        free      shared  buff/cache   available
Mem:       49281052     5365748    36426996       22664     7488308    42930820
Swap:      24772604           0    24772604
- DISK : 44Go Availabale

I did set this URL in the settings and restart : 
- nano ./settings/settings.py
`CORE_URI = "http://192.168.100.XX"`

Docker-compose ps Output : 
[docker-compose-ps.txt](https://github.com/mistio/mist-ce/files/2434684/docker-compose-ps.txt)

All I did find was this, in api container logs : 
[docker-compose-api-log.txt](https://github.com/mistio/mist-ce/files/2434683/docker-compose-api-log.txt)

Can you provide help ?

Best Regards,

Slevinox

**Comment by d-mo** [comment]
Hi Slevinox.

It looks like something is wrong with DNS resolution within the api container. The URI "http://landing" should be resolving to the landing container of mist. Instead of that, it looks like some local DNS server resolves it and directs it to an error page.

Can you try the following:
1. Restart the Docker daemon and see if that fixes things
2. If it doesn't, try editing /etc/resolv.conf in your server and enter `nameserver 8.8.8.8` in order to use the Google DNS server instead of your current one. I'm not sure if you need to restart the Docker daemon again for the changes to propagate.

**Comment by Slevinox** [comment]
Hi D-mo,

So restarting docker,
`systemctl restart docker`

And restarting all containers did not work
`docker-compose up -d`

I still got the same issue.

I did investigate docker's logs through and found this.
[docker-logs.txt](https://github.com/mistio/mist-ce/files/2437123/docker-logs.txt)
It seems that docker already use Google DNS.

**Comment by Slevinox** [comment]
After analyzing the api logs a second, I saw the DNS Resolve error message (sry i was a bit tired yesterday and missed it ^^').

So as I did set my server to use internal proxy, the request was send to it and it could not resolve "http://landing" if I am correct.

If so, I don't understand how it is working. I mean "landing" isn't supposed to be resolve directly inbetween containers composing mist using an internal DNS resolv configuration ?

Or did I missed something during configuration ?

**Comment by Slevinox** [comment]
Ok so after debugging and research to understand Docker handling of ENV var, I succeed.

So to resume the issue : 
- Containers were proxying requests to internal proxy causing the "Internal Server Error"

This were cause by envs var that was set due to multiple fails of several way of configuring docker proxy : 

1. ENV VAR : 
`export http_proxy=proxy_ip:proxy_port`
Did not help docker to work behind proxy
2. .docker/config.json
Did not help docker to work behind proxy and altered env var of running containers
3. /etc/systemd/system/docker.service.d/http-proxy.conf
create this file and put this in : 
`[Service]
Environment="HTTP_PROXY=proxy_ip:proxy_port"`

To correct this, I just deleted file of proxy "solution" 2 and everything work as expected now.

At last, sry @d-mo for your time and thanks a lot.
